### PR TITLE
OIDC RP stub

### PR DESCRIPTION
**Pull request by sanisc**
Dummy rp for OIDC provider to use

### CODE REVIEW

File: src/main/java/uk/gov/di/config/RelyingPartyConfig.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package uk.gov.di.config;
+
+public class RelyingPartyConfig {
+    public static final String CLIENT_ID="xpCRKVpmLFtDS3aOaiNAmsgnm2UNnEEh";
+    public static final String OP_AUTHORIZE_URL ="https://dev-g98b2qxy.eu.auth0.com/authorize";
```
Comment (mrwilson) [inline src/main/java/uk/gov/di/config/RelyingPartyConfig.java:5]: Are we happy to merge + deploy these real Auth0 urls, or can we set up the routes on `https://di-auth-oidc-provider.london.cloudapps.digital/` first and use those?

They could also be env-vars set up in the PaaS config.

Comment (sanisc) [reply src/main/java/uk/gov/di/config/RelyingPartyConfig.java:5]: This PR won't be merged as is - its just so there is a base and i agree we should point to the proper ui's in paas config and make the app default to localhost for local dev work

File: src/main/java/uk/gov/di/OidcRp.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+        InitRoutes();
+    }
+
+    public void InitRoutes(){
+        get("/", serveHomePage);
+
+        path("/oidc", () -> {
+            get("/auth", doAuthorise);
+            get("/callback", doAuthCallback);
```
Comment (JHjava) [inline src/main/java/uk/gov/di/OidcRp.java:21]: Might have missed it but I could find anything that triggered this `/callback`

Comment (JHjava) [reply src/main/java/uk/gov/di/OidcRp.java:21]: Ah ignore me, this is the redirect URI that we add to the auth request

File: build.gradle [hunk 3]
```diff
@@ -10,12 +10,14 @@ repositories {
 
 dependencies {
     compile(
-        "com.sparkjava:spark-core:2.9.3"
+        "com.sparkjava:spark-core:2.9.3",
+        "com.sparkjava:spark-template-mustache:2.7.1",
+        "com.nimbusds:oauth2-oidc-sdk:9.3.1"
     )
```
Comment (JHjava) [inline build.gradle:16]: You can enable logging by adding the `"org.slf4j:slf4j-simple:1.7.2"` dependency

File: src/main/java/uk/gov/di/config/RelyingPartyConfig.java (hunk 1 above)
Comment (sanisc) [reply src/main/java/uk/gov/di/config/RelyingPartyConfig.java]: updated config

**Review by JHjava — COMMENTED** [review]
Looks good. 

Really small comment, can we swap out the wildcard import in OidcRp? The GDS java style guide recommends against the use of wildcard imports.

**Comment by sanisc** [comment]
> Looks good.
> 
> Really small comment, can we swap out the wildcard import in OidcRp? The GDS java style guide recommends against the use of wildcard imports.

Done

**Review by JHjava — DISMISSED** [review]

**Review by JHjava — APPROVED** [review]

**Review by JHjava — APPROVED** [review]

**Pull request merged**